bpython/curtsies.py

```diff
@@ -122,6 +122,9 @@ def after_suspend():
                 window.__enter__()
                 interrupting_refresh()
 
+            def get_top_usable_line():
+                return window.top_usable_row
+
```

Comment:
As you point out, currently the Repl class doesn't have a handle on the Window object, all it has are callbacks for getting the information it needs.

You're suggesting a third class that inherits from the `curtsiesfrontend.repl.Repl` (which in turn inherits from the `repl.Repl`) that on instantiation is passed a window object and an input_generator (oh, or creating them itself would be better)? That sounds nice, I'm not sure why I did the callback style instead -- probably too much node.js and other less OOP programming recently. I don't see a reason not to, in fact the base curtsies repl class could even be functional like the old embedded version, and would be nice for testing.